src/SDK/SmartDetectors.Extensions/AlertExtensions.cs

```diff
@@ -195,23 +246,29 @@ private static AlertProperty CreateAlertProperty(Alert alert, PropertyInfo prope
         /// <param name="propertyName">The table property name.</param>
         /// <param name="displayName">The table property display name.</param>
         /// <param name="tableAttribute">The attribute applied to the table property.</param>
+        /// <param name="order">The order to use</param>
         /// <returns>The newly created <see cref="TableAlertProperty{T}"/> instance.</returns>
         private static DisplayableAlertProperty CreateTableAlertProperty(
             object propertyValue,
             string propertyName,
             string displayName,
-            TablePropertyAttribute tableAttribute)
+            TablePropertyAttribute tableAttribute,
+            Order order)
         {
             // Validate we have a proper value
             if (!(propertyValue is IList tablePropertyValue))
             {
                 throw new ArgumentException("An AlertPresentationTableAttribute can only be applied to properties of type IList");
             }
 
-            Type tableRowType = GetGenericListType(propertyValue.GetType());
-            if (tableRowType == null)
+            // Get element type, and verify that all elements are of the same type
+            Type tableRowType = tablePropertyValue[0].GetType();
```

Comment:
>tablePropertyValue[0].GetType(); [](start = 32, length = 32)

handle empty tables

Reply:
We won't get an empty table here - empty tables are handled by ExtractProperties. I added a check to make the exception clearer.

---
In reply to: [254177337](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/45#discussion_r254177337) [](ancestors = 254177337)